File: be/travelTourSem4-master/travelTourSem4-master/TravelTour/src/main/java/Sem4/TravelTour/entity/BookDetail.java
package Sem4.TravelTour.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@SuppressWarnings("serial")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "bookDetails")
public class BookDetail implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long bookDetailId;
    private int quantity;
    private Double price;
    private Date startDate;
    private Date endDate;

    @ManyToOne
    @JoinColumn(name = "tourId")
    private Tour tour;

    @ManyToOne
    @JoinColumn(name = "bookId")
    private Book book;

}

File: admin_fe/src/app/common/BookDetail.ts
import { Book } from "./Book";
import { Tour } from "./Tour";

export class BookDetail {
    'bookDetailId':number;
    'quantity':number;
    'price':number;
    'tour':Tour;
    'book':Book;
    'startDate':Date;
    'endDate':Date;
}
